network/ip.py

```diff
@@ -0,0 +1,615 @@
+              respectively whether the route should be set or removed.
+              or the interface should be brought up or down
+    addr:
+        required: false
+        aliases: [ net ]
+        description:
+            - The address to set or remove with prefix length. (e.g. 2001:db8::2/64, 10.0.0.2/24, 10.0.0.2/255.255.255.0)
+              For nets, the default route is specified by v4default or v6default .
+    dev:
+        required: true
+        description:
+            - The device to set or remove the address/route from. Or the name of the new link.
+    via:
+        required: false
+        description:
+            - IP address of gateway for manipulating routes.
+    kind:
+        required: false
+        choices: [ bridge, vlan, veth, port ]
+        description:
+            - Type of link to create or to set/remove interface as port of bridge.
+    link:
+        required: false
+        aliases: [ peer, master ]
+        description:
+            - The interface name of the parent link for vlan or second peer for veth interfaces.
+    vlan_id:
+        required: false
+        description:
+            - The vlan ID for vlan interfaces.
+'''
+
+EXAMPLES = '''
+# set address 2001:db8::42/64 to interface eth0
+- ip: mode=address addr=2001:db8::42/64 dev=eth0 state=present
+
+# change prefix length (watch out that any other assignments of same address with different prefix will be deleted)
+- ip: mode=address addr=2001:db8::42/112 dev=eth0 state=present
+
+# remove address from interface eth0
+- ip: mode=address addr=2001:db8::42/112 dev=eth0 state=absent
+
+# add route for net 2001:db3::/48 via fe80::3123 through interface eth0
+- ip: mode=route net=2001:db3::/48 via=fe80::3123 dev=eth0
+
+# add IPv4 default route through gateway 192.168.0.1 at eth0
+- ip: mode=route net=v4default via=192.168.0.1 dev=eth0
+
+# remove IPv6 default route at interface eth0
+- ip: mode=route net=::/0 dev=eth0 state=absent
+
+# create bridge br0
+- ip: mode=link dev=br0 kind=bridge
+
+# add eth0 to bridge br0
+- ip: mode=link dev=eth0 kind=port master=br0
+
+# remove eth0 from bridge
+- ip: mode=link dev=eth0 kind=port state=absent
+
+# create a veth interface pair of v1p0 and v1p1
+- ip: mode=link dev=v1p0 kind=veth peer=v1p1
+
+# add vlan interface with id 42 onto eth0
+- ip: mode=link dev=v100 kind=vlan link=eth0 vlan_id=42
+
+# delete interface br0
+- ip: mode=link dev=br0 state=absent 
+
+# bring up existing interface
+- ip: mode=link dev=eth1 
+
+# bring down pyhsical interface
+- ip: mode=link dev=eth1 state=absent 
+
+'''
+
+RETURN = '''
+---
+'''
+
+from ansible.module_utils.pycompat24 import get_exception
+
+try:
+    from pyroute2 import IPRoute
+    from pyroute2.netlink.exceptions import NetlinkError
+    from pyroute2.netlink import AF_INET6, AF_INET
+    import ipaddress
+    HAS_PYROUTE=1
+except ImportError:
+    HAS_PYROUTE=0
+
+ip=None
+module=None
+
+def l_key(l,key):
```

Comment:
The function should be more prominent in the documentation (http://docs.pyroute2.org/iproute.html) It is only used in one example on the top without description.